.github/workflows/container_build_template.yml

```diff
@@ -0,0 +1,193 @@
+---
+# this workflow file is the template
+# that will be triggered by a parent workflow matrix
+
+name: reusable container build workflow
+
+env:
+  # BUILDX_NO_DEFAULT_ATTESTATIONS must be set to build only arm64 and amd64 images
+  # the devcontainers/ci@v0.3 build will fail if this env variable is not set
+  BUILDX_NO_DEFAULT_ATTESTATIONS: 1
+
+on:
+  workflow_call:
+    inputs:
+      # container name, e.g. base, dev, etc.
+      container_name:
+        required: true
+        type: string
+      # container tags, e.g. "latest,python3.11" or "latest,python3.11-avd4.1.0"
+      # container tags are optional as they are auto-generated if not provided
+      container_tags:
+        required: false
+        type: string
+      # platform, e.g. linux/amd64, linux/arm64/v8
+      # default is just fine if there is no intention to release a special build
+      platform:
+        required: false
+        type: string
+        default: linux/arm64/v8,linux/amd64
+      # from_image, e.g. python, alpine, etc.
+      # currently we only support python-slim, this is a future knob
+      from_image:
+        required: false
+        type: string
+        default: python
+      # from_variant, e.g. 3.11-slim-bullseye
+      # this is auto-generated for python-slim based builds
+      # only specify for special builds
+      from_variant:
+        required: false
+        type: string
+      username:
+        required: false
+        type: string
+        default: avd
+      user_id:
+        required: false
+        type: string
+        default: 1000
+      group_id:
+        required: false
+        type: string
+        default: 1000
+      python_version:
+        required: true
+        type: string
+
+jobs:
+  build_image:
+    runs-on: ubuntu-22.04
+
+    steps:
+
+      - name: Starting container build
+        run: echo "Starting container build. Be patient. 🐢"
+
+      - name: Checkout code ✅
+        uses: actions/checkout@v3
```

Comment:
please use latest version (v4 today) in new workflows (we know we need to update the old ones ;) )

Reply:
OK. I have not tested this one yet. Changing. We'll see if it works as expected during container build.